modules/apps/29-fee/ibc_module_test.go

```diff
@@ -468,3 +470,301 @@ func (suite *FeeTestSuite) TestOnChanCloseConfirm() {
+				fungibleTokenPacket.GetBytes(),
+				suite.chainA.SenderAccount.GetSequence(),
+				suite.path.EndpointA.ChannelConfig.PortID,
+				suite.path.EndpointA.ChannelID,
+				suite.path.EndpointB.ChannelConfig.PortID,
+				suite.path.EndpointB.ChannelID,
+				clienttypes.NewHeight(0, 100),
+				0,
+			)
+
+			// set up & escrow packet fees
+			refundAcc := suite.chainA.SenderAccount.GetAddress()
+			packetId := channeltypes.NewPacketId(suite.path.EndpointA.ChannelID, suite.path.EndpointA.ChannelConfig.PortID, 1)
+			identifiedFee := types.NewIdentifiedPacketFee(
+				packetId,
+				types.Fee{
+					ReceiveFee: validCoins,
+					AckFee:     validCoins2,
+					TimeoutFee: validCoins3,
+				},
+				refundAcc.String(),
+				[]string{},
+			)
+			err := suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), identifiedFee)
+			suite.Require().NoError(err)
+
+			// set up module and callbacks
+			module, _, err := suite.chainA.App.GetIBCKeeper().PortKeeper.LookupModuleByPort(suite.chainA.GetContext(), ibctesting.TransferPort)
+			suite.Require().NoError(err)
+
+			cbs, ok := suite.chainA.App.GetIBCKeeper().Router.GetRoute(module)
+			suite.Require().True(ok)
+
+			// malleate test case
+			tc.malleate(suite)
+
+			result := cbs.OnRecvPacket(suite.chainA.GetContext(), packet, refundAcc)
+
+			if tc.sourceRelayer {
+				suite.Require().NoError(err, "unexpected error for case: %s", tc.name)
+				ack := types.IncentivizedAcknowledgement{
+					Result:                channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+					ForwardRelayerAddress: suite.chainB.SenderAccount.GetAddress().String(),
+				}
+				suite.Require().Equal(result, ack)
+			} else {
+				suite.Require().NoError(err, "unexpected error for case: %s", tc.name)
+				ack := types.IncentivizedAcknowledgement{
+					Result:                channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+					ForwardRelayerAddress: "",
+				}
+				suite.Require().Equal(result, ack)
+			}
+		})
+	}
+}
+
+func (suite *FeeTestSuite) TestOnAcknowledgementPacket() {
+	testCases := []struct {
+		name     string
+		malleate func(suite *FeeTestSuite)
+		expPass  bool
+	}{
+		{
+			"success",
+			func(suite *FeeTestSuite) {
+				packetId := channeltypes.NewPacketId(suite.path.EndpointA.ChannelID, suite.path.EndpointA.ChannelConfig.PortID, suite.chainA.SenderAccount.GetSequence())
+				identifiedFee := types.NewIdentifiedPacketFee(
+					packetId,
+					types.Fee{
+						ReceiveFee: validCoins,
+						AckFee:     validCoins2,
+						TimeoutFee: validCoins3,
+					},
+					suite.chainA.SenderAccount.GetAddress().String(),
+					[]string{},
+				)
+				err := suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), identifiedFee)
+				suite.Require().NoError(err)
+			},
+			true,
+		},
+		{
+			"identified packet fee doesn't exist",
+			func(suite *FeeTestSuite) {
+				suite.Require().Error(types.ErrFeeNotFound)
```

Comment:
This isn't actually testing the `OnAcknowledgementPacket` call, it's just testing your setup functions.

Reply:
Ahhh, I think see what you're doing here. The setup in `suite.run` fails when you don't explicitly escrow the packet etc in each test case. I think it would be easier to read (and also reduce code)  if you called `EscrowPacketFee` after calling malleate and used each case to malleate/change the data to force the test to fail :)

Reply:
it's not really that the setup fails but that the OnAck callback checks if an identified fee exists, so if it hasn't been escrowed then that error should surface

Reply:
Yeah, I see. I feel like it's a bit hard to read using the malleate functions like this (I think the general approach is you change (malleate) data here to force an error rather than have the default case fail) but I think it's ok for now :)

Reply:
I don't understand the purpose of `suite.Require().Error(types.ErrFeeNotFound)`. It is just asserting that `types.ErrFeeNotFound` is of type error, it isn't checking any logic related to ibc_module.go